gateway/1.4/modules/ROOT/pages/_partials/task-node-affinity.adoc

```diff
@@ -0,0 +1,238 @@
+
+[[list_add_labels]]
+== List and Create Node Labels
+
+Each node affinity rule in the Pod spec requires the key of a node label and one or more of the label’s values (see <<set_node_affinity>>). You can use existing node labels or create new ones. A node label is defined as a key-value pair.
+
+* To list a cluster's nodes and node labels, run: 
++
+[source,kubernetes,subs=attributes+]
+----
+kubectl get nodes --show-labels
+----
+
+* To create a label for a node in your cluster, run:
++
+[source,kubernetes,subs=attributes+]
+----
+kubectl label nodes <node-name> <label-key>=<label-value>
+----
+
+For related Kubernetes documentation, see https://kubernetes.io/docs/tasks/configure-pod-container/assign-pods-nodes/#add-a-label-to-a-node[Add a Label to a Node^].
+
+[[set_node_affinity]]
+== Set Node Affinity for a Pod
+
+The Kubernetes scheduler can use node affinity rules to match nodes to Pods. The matching process enables the scheduler to select the node into which a Pod is deployed.
+
+Configure node affinity rules within a Pod spec. A Pod spec is YAML file in which you can specify rules for required, preferred, or both types of matching:
+
+* `requiredDuringSchedulingIgnoredDuringExecution`: The node must match all node affinity rules configured for a Pod when the Kubernetes scheduler applies this type of matching. Otherwise, the deployment fails. 
+* `preferredDuringSchedulingIgnoredDuringExecution`: Based on weighted preferences set for a Pod, the Kubernetes scheduler selects the best matching node. The scheduler must select a node, so if no node matches any of the preferences, the scheduler selects a node that does not match. 
+
+For guidance, see the following configuration options:
+
+* <<strict_match_only>>
+* <<soft_match_only>>
+* <<all_match_types>>
+
+[[strict_match_only]]
+=== Configure a Required Node Matching Rule for a Pod 
+
+In required node matching, the node must meet all node affinity rules configured for a Pod. Otherwise, the deployment fails. 
+
+//end::temp-tag1[]
+
+//TAG for common node affinity settings
+//tag::config-flex-affinity-condition[]
+Configure a node affinity rule in the Pod spec file. Each condition to a rule requires a node label as a `key`, an operator, and one or more node label values to accept for a node with that key. For operators, you can use `In`, `NotIn`, `Exists`,  `DoesNotExist`, `Gt`, and `Lt`.
+//end::config-flex-affinity-condition[]
+
+[source,yaml,subs=attributes+]
+----
+affinity:
+ nodeAffinity:
+   requiredDuringSchedulingIgnoredDuringExecution:
+     nodeSelectorTerms:
+     - matchExpressions:
+       - key: <node-label-key>
+         operator: <operator>
+         values:
+         - <node-label-value>
+       # - Add any other node label values. 
+    # - Add any other required conditions.
+----
+
```

Comment:
Do we want to provide a more concrete or realistic example? And would it be more common to use required, preferred, or combined (required and fixed) node affinity for a flex pod? Or it just depends?

Reply:
A concrete example could be useful, but IMO this example is clear enough, this configuration is not very complex

Reply:
Regarding of what is more common to use, it really depends on your needs. If you need flex to run on specific nodes, then required is your way to go. If not maybe preferred is better suited

Reply:
Ok, sounds good.